### PR TITLE
HODS-76 | Added cache, integration tests

### PR DESCRIPTION
**Pull request by owen-opencast**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import uk.gov.hmrc.DefaultBuildSettings.integrationTestSettings
+
+//import uk.gov.hmrc.ServiceManagerPlugin.Keys.itDependenciesList
```
Comment (parkeradam) [inline build.sbt:3]: Please remove commented out dependencies

File: test/it/uk/gov/hmrc/organisationsmatchingapi/services/CacheServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import uk.gov.hmrc.organisationsmatchingapi.models.{Address, CrnMatch, CrnMatchingRequest, SaUtrMatch, SaUtrMatchingRequest}
+import uk.gov.hmrc.organisationsmatchingapi.services.CacheService
+import util.UnitSpec
+
+import scala.concurrent.Future
+
+class CacheServiceSpec extends UnitSpec with Matchers with GuiceOneAppPerSuite with MockitoSugar with ScalaFutures with BeforeAndAfterEach {
+
+
+  override def beforeEach() {
```
Comment (parkeradam) [inline test/it/uk/gov/hmrc/organisationsmatchingapi/services/CacheServiceSpec.scala:36]: Might not need to extende BeforeAndAfterEach and remove this empty override?

**Review by parkeradam — CHANGES_REQUESTED** [review]
Also unsure why Documentation Controller has been purged as this will break the pipelines for publishing. 
Also it:test will need adding to the job in build-jobs repo.

File: app/uk/gov/hmrc/organisationsmatchingapi/platform/controllers/DocumentationController.scala [hunk 3]
```diff
@@ -1,47 +0,0 @@
-/*
```
Comment (ajchilts) [inline app/uk/gov/hmrc/organisationsmatchingapi/platform/controllers/DocumentationController.scala:1]: This file is needed for the publish. Please can we add it back in?

File: build.sbt (hunk 1 above)
Comment (owen-opencast) [reply build.sbt:3]: 👍

File: app/uk/gov/hmrc/organisationsmatchingapi/services/MatchingService.scala [hunk 4]
```diff
@@ -1,45 +0,0 @@
-/*
```
Comment (ajchilts) [inline app/uk/gov/hmrc/organisationsmatchingapi/services/MatchingService.scala:1]: I assume we are deleting this such that we can write our own implementation later?

File: test/it/uk/gov/hmrc/organisationsmatchingapi/services/CacheServiceSpec.scala (hunk 2 above)
Comment (ajchilts) [reply test/it/uk/gov/hmrc/organisationsmatchingapi/services/CacheServiceSpec.scala:36]: I think this is needed and the implementation here is simply overriding the `out of the box` logic?

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/organisationsmatchingapi/platform/controllers/DocumentationController.scala (hunk 3 above)
Comment (owen-opencast) [reply app/uk/gov/hmrc/organisationsmatchingapi/platform/controllers/DocumentationController.scala:1]: I removed the redundant DocumentationController that had been left in from the original repo - there’s still one in there, the one referred to in the routes. Can we recheck?

File: app/uk/gov/hmrc/organisationsmatchingapi/services/MatchingService.scala (hunk 4 above)
Comment (owen-opencast) [reply app/uk/gov/hmrc/organisationsmatchingapi/services/MatchingService.scala:1]: Yes - it was just duplicating the CacheService, it didn't have any implementation in it

**Review by ajchilts — APPROVED** [review]
When the other comments addressed :-)

File: test/it/uk/gov/hmrc/organisationsmatchingapi/services/CacheServiceSpec.scala (hunk 2 above)
Comment (owen-opencast) [reply test/it/uk/gov/hmrc/organisationsmatchingapi/services/CacheServiceSpec.scala:36]: Removed - don't think it's needed

**Review by parkeradam — APPROVED** [review]

**Pull request merged**